### PR TITLE
RAS-491: Add session caching functionality 

### PR DESCRIPTION
**Pull request by RaminEsfandiari**

### CODE REVIEW

File: app/controllers/MemberDetailsController.scala [hunk 1]
```diff
@@ -28,36 +28,39 @@ import uk.gov.hmrc.time.TaxYearResolver
...
-      Future.successful(Ok(views.html.member_details(form)))
-    case Left(resp) =>  Logger.debug("[MemberDetailsController][get] user Not authorised"); resp
-    }
+  def get = Action.async {
+    implicit request =>
+      isAuthorised.flatMap {
+        case Right(userInfo) =>
+          Logger.debug("[MemberDetailsController][get] user authorised")
+          sessionService.fetchMemberDetails() map {
+            case Some(md) => Ok(views.html.member_details(form.fill(md)))
```
Comment (joepowell96) [inline app/controllers/MemberDetailsController.scala]: Change md to something more meaningful

File: app/controllers/MemberDetailsController.scala [hunk 2]
```diff
@@ -69,61 +72,62 @@ trait MemberDetailsController extends RasController {
...
+              }
+
+              residencyStatusAPIConnector.getResidencyStatus(uuid.get).map { rasResponse =>
+
+                val name = memberDetails.firstName + " " + memberDetails.lastName
+                val dateOfBirth = memberDetails.dateOfBirth.asLocalDate.toString("d MMMM yyyy")
+                val cyResidencyStatus = extractResidencyStatus(rasResponse.currentYearResidencyStatus)
+                val nyResidencyStatus = extractResidencyStatus(rasResponse.nextYearForecastResidencyStatus)
+
+                if (cyResidencyStatus.isEmpty || nyResidencyStatus.isEmpty) {
```
Comment (joepowell96) [inline app/controllers/MemberDetailsController.scala]: Is this what we want to do?

If we get current year should we just return current year

Comment (RaminEsfandiari) [reply app/controllers/MemberDetailsController.scala]: Will do this as part of RAS-474

Comment (joepowell96) [reply app/controllers/MemberDetailsController.scala]: Leave for now, more a query to discuss.

**Pull request merged**